src/Psalm/Internal/PhpVisitor/Reflector/ExpressionResolver.php

```diff
@@ -445,6 +445,38 @@ private static function functionEvaluatesToTrue(
             return false;
         }
 
+        if ($function->name->parts === ['enum_exists']
+            && isset($function->getArgs()[0])
+        ) {
+            $string_value = null;
+
+            if ($function->getArgs()[0]->value instanceof PhpParser\Node\Scalar\String_) {
+                $string_value = $function->getArgs()[0]->value->value;
+            } elseif ($function->getArgs()[0]->value instanceof PhpParser\Node\Expr\ClassConstFetch
+                && $function->getArgs()[0]->value->class instanceof PhpParser\Node\Name
+                && $function->getArgs()[0]->value->name instanceof PhpParser\Node\Identifier
+                && strtolower($function->getArgs()[0]->value->name->name) === 'class'
+            ) {
+                $string_value = (string) $function->getArgs()[0]->value->class->getAttribute('resolvedName');
+            }
+
+            // We're using class_exists here because enum_exists doesn't exist on old versions of PHP
+            // Not sure what happens if we try to autoload or reflect on an enum on an old version of PHP though...
+            if ($string_value && class_exists($string_value)) {
+                $reflection_class = new ReflectionClass($string_value);
```

Comment:
Well, it will only work if PHP was able to autoload the file and parse it and detect it is indeed an Enum. Which mean it will only work on PHP >8.1. At this point, we might as well use enum_exists as long as we checked we're in 8.1.

Thinking about it however, I don't see a point preventing scanning of a possible class that would have been pushed in an enum_exists. Either it exists and it's useful and it will end up scanned at one point or it's useless and it's possible Psalm will actually report it unused

Anyway, the real issue is for older versions of PHP, I have no idea what will happen if we run Psalm on 8.0 and try to analyze an enum. It will probably crash? But I don't think I can prevent it